### PR TITLE
Split Track.cpp and Track.h

### PR DESCRIPTION
**Pull request by M374LX**
This PR splits Track.cpp and Track.h into one pair of files per class, as suggested in #5592.

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
@M374LX This is just moving code between files, nothing else?

**Comment by M374LX** [comment]
@JohannesLorenz Other than moving code between files, there are a few changes in includes and forward declarations.

**Comment by JohannesLorenz** [comment]
Are you able to rebase this branch, so the mentioned conflicts will vanish?

**Comment by JohannesLorenz** [comment]
Let me know when I should review it.

Also, as you develop more PRs, just as information: We are currently trying to stop writing new PRs, in order to get our already started PRs closed for the reorganizing (#5592).

**Comment by M374LX** [comment]
@JohannesLorenz It is ready for review.

File: src/gui/TrackView.cpp [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include "TrackView.h"
+
+#include <assert.h>
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:28]: Can you please replace this by including `QtGlobal` and change `assert` below to `Q_ASSERT`?

File: src/gui/TrackView.cpp [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+#include <cstdlib>
+
+#include <QLayout>
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:31]: not used

File: src/gui/TrackView.cpp [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+#include <QLinearGradient>
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:32]: not used

File: src/gui/TrackView.cpp [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+#include <QMenu>
+#include <QMouseEvent>
+#include <QPainter>
+#include <QStyleOption>
+#include <QVariant>
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:37]: not used

File: src/gui/TrackView.cpp [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+#include <QClipboard>
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:38]: not used

File: src/gui/TrackView.cpp [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+
+
+#include "AutomationPattern.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:41]: All until (and including) BBTrackContainer.h are not used.

File: src/gui/TrackView.cpp [hunk 7]
```diff
@@ -0,0 +1,473 @@
...
+#include "AutomationTrack.h"
+#include "AutomationEditor.h"
+#include "BBEditor.h"
+#include "BBTrack.h"
+#include "BBTrackContainer.h"
+#include "ConfigManager.h"
+#include "Clipboard.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:48]: not used

File: src/gui/TrackView.cpp [hunk 8]
```diff
@@ -0,0 +1,473 @@
...
+#include "ColorChooser.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:49]: not used

File: src/gui/TrackView.cpp [hunk 9]
```diff
@@ -0,0 +1,473 @@
...
+#include "embed.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:50]: not used

File: src/gui/TrackView.cpp [hunk 10]
```diff
@@ -0,0 +1,473 @@
...
+#include "Engine.h"
+#include "GuiApplication.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:52]: not used

File: src/gui/TrackView.cpp [hunk 11]
```diff
@@ -0,0 +1,473 @@
...
+#include "FxMixerView.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:53]: not used

File: src/gui/TrackView.cpp [hunk 12]
```diff
@@ -0,0 +1,473 @@
...
+#include "gui_templates.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:54]: not used

File: src/gui/TrackView.cpp [hunk 13]
```diff
@@ -0,0 +1,473 @@
...
+#include "MainWindow.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:55]: not used

File: src/gui/TrackView.cpp [hunk 14]
```diff
@@ -0,0 +1,473 @@
...
+#include "Mixer.h"
+#include "ProjectJournal.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:57]: All until (and including) SongEditor.h are not used

File: src/gui/TrackView.cpp [hunk 15]
```diff
@@ -0,0 +1,473 @@
...
+#include "SampleTrack.h"
+#include "Song.h"
+#include "SongEditor.h"
+#include "StringPairDrag.h"
+#include "TextFloat.h"
```
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:62]: Not used

File: src/gui/widgets/FxLineLcdSpinBox.cpp [hunk 16]
```diff
@@ -28,6 +28,7 @@
 #include "FxMixerView.h"
 #include "GuiApplication.h"
 #include "Track.h"
```
Comment (JohannesLorenz) [inline src/gui/widgets/FxLineLcdSpinBox.cpp:30]: can be removed

**Review by JohannesLorenz — CHANGES_REQUESTED** [review]
Very useful PR, thanks for your efforts.

I left a lot of comments, but it's almost all about unused includes. Removing them will partly be a bit of work, but it will increase compile time.

File: include/BBEditor.h [hunk 17]
```diff
@@ -26,8 +26,10 @@
 #ifndef BB_EDITOR_H
 #define BB_EDITOR_H
 
+
 #include "Editor.h"
 #include "TrackContainerView.h"
+#include "TrackContentObjectView.h"
```
[the remark this answers is not in the document]
Comment (M374LX) [reply include/BBEditor.h:32]: That is right. I should have added it to TrackContentWidget.cpp instead.

File: src/gui/TrackView.cpp (hunk 14 above)
Comment (M374LX) [reply src/gui/TrackView.cpp:57]: Mixer is actually needed, as seen in ```TrackView::dropEvent()```.

Comment (JohannesLorenz) [reply src/gui/TrackView.cpp:57]: Good point :+1:

File: include/TrackContainerView.h [hunk 18]
```diff
@@ -34,6 +34,7 @@
 #include "Track.h"
 #include "JournallingObject.h"
 #include "InstrumentTrack.h"
+#include "Rubberband.h"
```
[the remark this answers is not in the document]
Comment (M374LX) [reply include/TrackContainerView.h:36]: This one gave me a lot of work, but I did it. :+1:

File: src/core/Track.cpp [hunk 19]
```diff
@@ -37,2405 +37,18 @@
```
Comment (JohannesLorenz) [inline src/core/Track.cpp:37]: I did not see it last review - Can you please fix the top comment (lines 2 and 3) and the doxygen `\file` comment (line 27)?

**Review by JohannesLorenz — APPROVED** [review]
Only one review comment added about some code comments. Functional it's all OK.

**Review by JohannesLorenz — APPROVED** [review]
LGTM!

**Comment by JohannesLorenz** [comment]
Thanks for the PR!

I'll wait for CI to finish and then I will "Squash and Merge". If you think it's too early for a merge because you want to add more commits for any reason, scream now :smiley:

**Comment by M374LX** [comment]
@JohannesLorenz I believe it is ready to be merged.

**Pull request merged**